graphql_api/tests/test_impacted_file_new.py

```diff
@@ -0,0 +1,846 @@
+        )
+        self.pull = PullFactory(
+            pullid=44,
+            repository=self.commit.repository,
+            head=self.commit.commitid,
+            base=self.parent_commit.commitid,
+            compared_to=self.parent_commit.commitid,
+        )
+        self.comparison = CommitComparisonFactory(
+            base_commit=self.parent_commit,
+            compare_commit=self.commit,
+            state=CommitComparison.CommitComparisonStates.PROCESSED,
+            report_storage_path="v4/test.json",
+        )
+        self.comparison_report = ComparisonReport(self.comparison)
+
+    @patch("services.report.build_report_from_commit")
+    @patch("services.archive.ArchiveService.read_file")
+    @patch("services.comparison.Comparison.git_comparison")
+    def test_filtering_with_flags(
+        self, git_comparison_mock, read_file, build_report_from_commit
+    ):
+        git_comparison_mock.return_value = None
+        build_report_from_commit.return_value = sample_report()
+        read_file.return_value = mock_data_from_archive
+
+        query_impacted_files = """
+        query ImpactedFilesNew(
+            $org: String!
+            $repo: String!
+            $commit: String!
+            $filters: ImpactedFilesFilters
+        ) {
+            owner(username: $org) {
+                repository(name: $repo) {
+                    ... on Repository {
+                        commit(id: $commit) {
+                            compareWithParent {
+                                ... on Comparison {
+                                    impactedFilesNew(filters: $filters) {
+                                        ... on ImpactedFilesNew {
+                                            results {
+                                                fileName
+                                            }
+                                        }
+                                        ... on UnknownFlags {
+									        message
+								        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        """
+
+        variables = {
+            "org": self.org.username,
+            "repo": self.repo.name,
+            "commit": self.commit.commitid,
+            "filters": {"flags": ["fake_flag"]},
+        }
+
+        data = self.gql_request(query_impacted_files, variables=variables)
+        assert data["owner"]["repository"]["commit"]["compareWithParent"][
+            "impactedFilesNew"
+        ] == {"message": "No coverage with chosen flags"}
+
+        variables = {
+            "org": self.org.username,
+            "repo": self.repo.name,
+            "commit": self.commit.commitid,
+            "filters": {"flags": ["flag1"]},
+        }
+
+        data = self.gql_request(query_impacted_files, variables=variables)
+        assert (
```

Comment:
Lines 328 onwards feels like another test, so like renaming this to "unknown flags" and the next one to "successful test", and assert exactly the response we get